### ISSUE TITLE
Cannot get it work

### ISSUE THREAD

**Issue by chansters** [rs1932/homeassistant-ring_alarm_component#1]
I followed all the installation steps but it doesn't seem to work.  I don't see any new entities or sensors created ? I'm running Home Assistant as a docker in Unraid. 

The component seems to be loading correctly as I see this in the log : 

"2019-09-09 16:53:04 WARNING (MainThread) [homeassistant.loader] You are using a custom integration for ring_alarm which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you do experience issues with Home Assistant." 

But I do see this in the log : "[engineio.client] Received unexpected packet of type 2".  

Thanks !

**Comment by rs1932** [comment]
I have some logging enabled in the component. Do you see any INFO messages on this custom component? If you don't see any, you might need to enable INFO logging in the configuration.yaml file. It would report it was able to log in. the ring alarm site and start getting information.

**Comment by chansters** [comment]
The component seems to be registering fine but I don't see any sensors that I can work with? I don't see any new entities in the States view. 

Here are all the ring alarm related messages that I can see in the log : 

2019-09-13 16:50:50 WARNING (MainThread) [homeassistant.loader] You are using a custom integration for ring_alarm which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you do experience issues with Home Assistant.

2019-09-13 16:50:52 INFO (MainThread) [homeassistant.setup] Setting up ring_alarm

2019-09-13 16:50:54 INFO (SyncWorker_0) [custom_components.ring_alarm] The ring alarm component is being initialized!

2019-09-13 16:50:55 INFO (SyncWorker_0) [custom_components.ring_alarm.pyringalarm] Oauth Token obtained

2019-09-13 16:50:55 INFO (SyncWorker_0) [custom_components.ring_alarm.pyringalarm] User locations are *took the rest of this out*

2019-09-13 16:50:56 INFO (SyncWorker_0) [custom_components.ring_alarm.pyringalarm] Total hubs 1

2019-09-13 16:50:56 INFO (SyncWorker_0) [custom_components.ring_alarm] Getting ring alarm devices information

2019-09-13 16:50:56 INFO (MainThread) [homeassistant.setup] Setup of domain ring_alarm took 4.1 seconds.

2019-09-13 16:50:57 INFO (Thread-10) [custom_components.ring_alarm] Total number of Ring Alarm devices: 16

**Comment by rs1932** [comment]
Ok, that means the component is working and is able to get the devices and the information. Have looked at the entities and seen if it exists there (developer tools/states)?

**Comment by chansters** [comment]
No, there are no components in the States view that relate to Ring Alarm.

But I see this error message that only appears when the Ring Alarm component is used : 

 "[engineio.client] Received unexpected packet of type 2".

When I Googled this error message, it seems to relate to Python Socket IO ?

**Comment by rs1932** [comment]
Yes the packet of type 2 appears periodically for me as well. Its not connected to what ur facing.

After this "2019-09-13 16:50:57 INFO (Thread-10) [custom_components.ring_alarm] Total number of Ring Alarm devices: 16"

do u see any more messages - I have INFO messages for every device added for example:
2019-09-07 23:28:32 INFO (Thread-9) [custom_components.ring_alarm] Adding component front tree floodlights

What kind of devices do u have?

**Comment by chansters** [comment]
I don't have any messages that call out specific Ring alarm devices unlike yours. I don't have anything fancy.  Just the Ring Alarm hub, window and door sensors and motion sensors.  I'm actually surprised it found 16 items because I don't think I have that many.  I just have what came with the standard Ring Alarm kit.  

It's seems to be communicating with Ring Alarm.  I just went out and came home (alarm, un-alarmed etc) and now there are quite a few rows of this : 

2019-09-13 17:31:21 INFO (Thread-11) [custom_components.ring_alarm] Update received for device 66ec16f3-3d9b-4136-be26-48d98e02fce9
2019-09-13 17:34:31 INFO (Thread-12) [custom_components.ring_alarm] Update received for device 9fb1f312-927e-4635-8487-f17813d36a47
2019-09-13 17:35:40 INFO (Thread-13) [custom_components.ring_alarm] Update received for device 9fb1f312-927e-4635-8487-f17813d36a47
2019-09-13 17:35:49 INFO (Thread-14) [custom_components.ring_alarm] Update received for device 9c133d19-a5fb-44a5-a46e-7575a1d1849a
2019-09-13 17:36:39 INFO (Thread-15) [custom_components.ring_alarm] Update received for device f4b4e2cf-7aec-4842-93d2-fb6580591529

I don't know whether it's a factor but I also have the Ring Doorbell which is working fine.

**Comment by rs1932** [comment]
Thats strange, I have the same ring alarm you have and the lighting hub as well. From the messages it is able to communicate with Ring and is subscribed to messages over the web socket from there. 

Just to confirm you don't see entities like these " binary_sensor.back_door"  or "binary_sensor.hallway_motion_sensor" the portion after "binary_sensor" should be what you named ur sensor?

Let me add a couple of addition info statements and put out a new version for you to test.

**Comment by dshokouhi** [comment]
Odd I see the same error as OP does however all of my expected binary sensors and switches are created.  I have a ring smart lighting system with Pathlights, Floodlight battery, Spotlights and a Motion Sensor.  All showing the proper data.

**Comment by chansters** [comment]
@rs1932 - Thanks ! Would love to try anything that could help me figure out what’s going on. 

@dshokouhi - Very strange ! I’m running Home Assistant as a Docker app within Unraid.  What’s your HA installation ?

**Comment by dshokouhi** [comment]
@chansters venv in ubuntu

**Comment by rs1932** [comment]
@chansters : Sorry couldn't get to this earlier. I released a debug version for you to try. The only difference is I added a statement to print out the contents of the device for binary sensor that are returned from ring. just trying to narrow. Once u run this it should print out a bunch of information on the devices. lets  see if it reaches there.

**Comment by chansters** [comment]
No worries ! Thanks for doing this ! I installed the debug version but unfortunately I don't see any new messages in the log.  And still don't see any ring alarm entities created. :(

2019-09-21 00:27:25 WARNING (MainThread) [homeassistant.loader] You are using a custom integration for ring_alarm which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you do experience issues with Home Assistant.

2019-09-21 00:27:27 INFO (MainThread) [homeassistant.setup] Setting up ring_alarm

2019-09-21 00:27:30 INFO (SyncWorker_1) [custom_components.ring_alarm] The ring alarm component is being initialized!

2019-09-21 00:27:30 INFO (SyncWorker_1) [custom_components.ring_alarm.pyringalarm] Oauth Token obtained

2019-09-21 00:27:31 INFO (SyncWorker_1) [custom_components.ring_alarm.pyringalarm] User locations are [*removed*]

2019-09-21 00:27:32 INFO (SyncWorker_1) [custom_components.ring_alarm.pyringalarm] Total hubs 1

2019-09-21 00:27:32 INFO (SyncWorker_1) [custom_components.ring_alarm] Getting ring alarm devices information

2019-09-21 00:27:32 INFO (MainThread) [homeassistant.setup] Setup of domain ring_alarm took 4.6 seconds.

2019-09-21 00:27:32 INFO (Thread-8) [custom_components.ring_alarm] Total number of Ring Alarm devices: 16

2019-09-21 00:27:38 INFO (SyncWorker_12) [homeassistant.loader] Loaded alarm_control_panel from homeassistant.components.alarm_control_panel

2019-09-21 00:27:38 INFO (MainThread) [homeassistant.setup] Setting up alarm_control_panel

2019-09-21 00:27:38 INFO (MainThread) [homeassistant.setup] Setup of domain alarm_control_panel took 0.0 seconds.

2019-09-21 00:27:38 INFO (MainThread) [homeassistant.components.alarm_control_panel] Setting up alarm_control_panel.wink

2019-09-21 00:31:22 INFO (Thread-14) [custom_components.ring_alarm] Update received for device 66ec16f3-3d9b-4136-be26-48d98e02fce9

2019-09-21 00:33:56 INFO (Thread-19) [custom_components.ring_alarm] Update received for device 9c133d19-a5fb-44a5-a46e-7575a1d1849a

**Comment by rs1932** [comment]
Strange, I have this running under Unraid as well.

I just put out a new debug version,  _debug_v03 - please try this version.
Right after the "Total number of Ring Alarm devices: " message you should see this 
"Raw List of all devices obtained from Ring"
followed a list of your Ring device names.

Lets see if we can atleast see the names of the ring devices.

**Comment by chansters** [comment]
Thanks again but I don't know why it's not printing the name of the devices.  :(.  I added "ABC" to the start of this line (_LOGGER.info("ABCTotal number of Ring Alarm devices: " + str(len(entities_data.index)))) because I wanted to confirm that it was loading the latest init.py file and it seems to be.  But for some reason, the list of devices doesn't show. 

Here's the latest log 

2019-09-21 23:34:41 WARNING (MainThread) [homeassistant.loader] You are using a custom integration for ring_alarm which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you do experience issues with Home Assistant.

2019-09-21 23:34:41 INFO (MainThread) [homeassistant.setup] Setting up ring_alarm

2019-09-21 23:34:41 INFO (SyncWorker_3) [custom_components.ring_alarm] The ring alarm component is being initialized!

2019-09-21 23:34:42 INFO (SyncWorker_3) [custom_components.ring_alarm.pyringalarm] Oauth Token obtained

2019-09-21 23:34:42 INFO (SyncWorker_3) [custom_components.ring_alarm.pyringalarm] User locations are *removed*

2019-09-21 23:34:43 INFO (SyncWorker_3) [custom_components.ring_alarm.pyringalarm] Total hubs 1

2019-09-21 23:34:43 INFO (SyncWorker_3) [custom_components.ring_alarm] Getting ring alarm devices information

2019-09-21 23:34:43 INFO (MainThread) [homeassistant.setup] Setup of domain ring_alarm took 1.8 seconds.

2019-09-21 23:34:43 INFO (SyncWorker_6) [homeassistant.loader] Loaded alarm_control_panel from homeassistant.components.alarm_control_panel

2019-09-21 23:34:43 INFO (MainThread) [homeassistant.setup] Setting up alarm_control_panel

2019-09-21 23:34:43 INFO (MainThread) [homeassistant.setup] Setup of domain alarm_control_panel took 0.0 seconds.

2019-09-21 23:34:43 INFO (MainThread) [homeassistant.components.alarm_control_panel] Setting up alarm_control_panel.wink

2019-09-21 23:34:44 INFO (Thread-8) [custom_components.ring_alarm] ABCTotal number of Ring Alarm devices: 16

2019-09-21 23:41:29 INFO (Thread-20) [custom_components.ring_alarm] Update received for device 01f5ab4e-fc41-4ea4-a937-76943989d246

2019-09-21 23:41:30 INFO (Thread-21) [custom_components.ring_alarm] Update received for device 01f5ab4e-fc41-4ea4-a937-76943989d246

**Comment by chansters** [comment]
So here's the weird thing.  I have been trying to get this work with HA running as a docker in Unraid.  Last night, I decided to run HA under Hass.io in Unraid. There's a Hass.io add on for Ring Alarm and I installed that.  And all my Ring Alarm devices show up when I do it this way ! 

Thank you so much for all your help with this but I will probably stick with the Hass.io implementation as it does seem to make it easier to implement some of the other things I want to do.

<img width="1173" alt="Home Assistant 2019-09-22 08-38-55" src="https://user-images.githubusercontent.com/8441205/65390501-75fb5200-dd14-11e9-8285-b5178f1d25c2.png">

**Comment by rs1932** [comment]
I did try this component running under the HA docker component under Unraid and it works fine, not sure why it doesn't work for you.
I have a Hass.io add-on for the ring alarm, possibly ur using that - its listed under my GitHub account. 

The reason I wrote this as a native component is that the add-on depends on a couple of other libraries written in JS, so the add on actually runs NodeJS as well. The biggest reason though was that it doesn't support the new Light component.

Glad that u got it working!
